executable-spec/bench/micro-benchmarking/Cardano/Ledger/MBenchmarks/Update/Tally/TallyMicroBenchmark.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE GADTs #-}
+
+module Cardano.Ledger.MBenchmarks.Update.Tally.TallyMicroBenchmark where
+
+import            Data.List (foldr, zip, repeat, take, map)
+import qualified  Data.Map.Strict as Map
+
+import           Ledger.Core (BlockCount (BlockCount), Slot (Slot),
+                     SlotCount (SlotCount), (*.), (+.))
+import           Cardano.Crypto.DSIGN.Mock (SignKeyDSIGN (SignKeyMockDSIGN),
+                     VerKeyDSIGN (VerKeyMockDSIGN))
+
+import            Cardano.Ledger.Spec.State.ProposalsState (ProposalsState (ProposalsState), tally)
+import            Cardano.Ledger.Spec.State.ProposalState ( ProposalState (ProposalState)
+                                                          , Decision (Rejected, NoQuorum, Expired, Accepted, Undecided)
+                                                          , VotingPeriod (VotingPeriod) 
+                                                          , decision
+                                                          )
+import           Cardano.Ledger.Spec.STS.Update.Data
+                     (Confidence (Abstain, Against, For), Stake (Stake))
+import           Cardano.Ledger.Spec.State.StakeDistribution (StakeDistribution (StakeDistribution))                                          
+import           Cardano.Ledger.Spec.Classes.Hashable (HasHash, Hash, Hashable,
+                     hash)
+import           Cardano.Ledger.Spec.Classes.HasSigningScheme (VKey)
+import           Cardano.Ledger.Test.Mock (Mock)
+
+
+data BenchmarkParams =
+  BenchmarkParams
+  { participants   :: Word
+  -- ^ Number of participants.
+  , concurrentSIPs :: Word
+  -- ^ SIP's that are active at the same time. We assume their voting period
+  -- overlaps exactly, which is the worst case.
+  }
+  deriving (Show, Eq)
+
+
+type TallyResult = [Decision]
+
+-- | Run the tally benchmark
+benchmarkTally = undefined
+
+-- | Get number of participants and run the tally
+runTally :: BenchmarkParams -> TallyResult
```

Comment:
This function does quite a lot of other stuff besides tallying. If we want to benchmark `tally` only we should do the setup before running the benchmarks.